e-shop-manager/src/test/java/it/unipd/mtss/business/BillImplTest.java

```diff
@@ -7,12 +7,38 @@
 
 import org.junit.Test;
 
-import it.unipd.mtss.business.exeption.EItemNotFoundException;
+import it.unipd.mtss.business.exception.BillException;
+import it.unipd.mtss.business.exception.EItemNotFoundException;
 import it.unipd.mtss.model.EItem;
 import it.unipd.mtss.model.EItemType;
+import it.unipd.mtss.model.User;
 
 public class BillImplTest {
 
+    @Test
+    public void testGetOrderPrice(){
+        BillImpl bill = new BillImpl();
+
+        List<EItem> items = new ArrayList<EItem>(List.of( 
+            new EItem(EItemType.Processor, "Processore1", 14.50),
+            new EItem(EItemType.Motherboard, "Scheda1", 12.50),
+            new EItem(EItemType.Processor, "Processore2", 10.50),
+            new EItem(EItemType.Processor, "Processore3", 6.50),
+            new EItem(EItemType.Keyboard, "Tastiera1", 16.50),
+            new EItem(EItemType.Mouse, "Mouse1", 3.50)
+        ));
+
+        User user = new User(1, "Mario", "Rossi", "1990-05-01");
+
+        try {
+            double price = bill.getOrderPrice(items, user);
+
+            assertEquals(64.0, price, 0);
+        } catch (BillException e) {
+            e.printStackTrace();
+        }
```

Comment:
Mi sono informato dicono di non mettere try, catch nel test, perchè qui tu stai testando un caso che sai già che non lancerà un eccezione. Devi dichiarare che il metodo rilancerà quell'eccezione nella firma e non ti da errore senza try catch, così: public void testGetOrderPrice throws BillException{...}
Poi devi fare un metodo apposito per testare il caso in cui succede l'eccezione, però qui non serve perchè non verrà mai lanciata.